### PR TITLE
ID-379 implement ToS Details endpoint and add new+existing endpoints to Swagger

### PR DESCRIPTION
**Pull request by tlangs**
Ticket: https://broadworkbench.atlassian.net/browse/ID-379

What:

This PR adds a new ToS Details endpoint. It returns whether ToS is enabled, if there's a grace period, and the current version of the ToS. I've also added this new endpoint and existing ToS endpoints to the Swagger config, for ease of development in the future.

Why:

We want the ToS experience in the Terra UI to be much better than it currently is. We want to be able to give users a warning that the ToS has been updated, give them a grace period to accept the new ToS, and make UI decisions about the ToS page/links. 

How:

Pretty simple PR. Just adding a new endpoint and updating Swagger. I've added a new test for the endpoint as well.

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://broadinstitute.github.io/dsp-appsec-security-risk-assessment/) and attached the result to the JIRA ticket

### CODE REVIEW

File: src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -2744,6 +2744,26 @@ paths:
...
                 $ref: '#/components/schemas/ErrorReport'
+  /register/user/v1/termsofservice/status:
+    get:
+      tags:
+        - Terms of Service
+      summary: returns whether or not the user has accepted the current terms of service.
+      operationId: getUserTermsOfServiceStatus
+      responses:
+        200:
+          description: terms of service is enabled and user
```
Comment (Shakespeared) [inline src/main/resources/swagger/api-docs.yaml]: I think you left out a couple words here!

**Review by Shakespeared — APPROVED** [review]
LGTM just noticed one typo

File: src/main/resources/swagger/api-docs.yaml [hunk 2]
```diff
@@ -2744,6 +2744,26 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                type: boolean
+                description: if terms of service is enabled and the user has accepted the latest version
```
Comment (Ghost-in-a-Jar) [inline src/main/resources/swagger/api-docs.yaml]: should this desc only be above on line 2754 (yikes this is a big file haha)

**Review by Ghost-in-a-Jar — APPROVED** [review]

**Comment by tlangs** [comment]
jenkins retest

**Comment by tlangs** [comment]
jenkins retest

**Pull request merged**